client/core/wallet.go

```diff
@@ -0,0 +1,107 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+package core
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"decred.org/dcrdex/client/asset"
+	"decred.org/dcrdex/dex"
+)
+
+// xcWallet is a wallet.
+type xcWallet struct {
+	asset.Wallet
+	connector *dex.ConnectionMaster
+	AssetID   uint32
+	mtx       sync.RWMutex
+	lockTime  time.Time
+	hookedUp  bool
+	balance   uint64
+	balUpdate time.Time
+	encPW     []byte
+	address   string
+}
+
+// Unlock unlocks the wallet.
+func (w *xcWallet) Unlock(pw string, dur time.Duration) error {
+	err := w.Wallet.Unlock(pw, dur)
+	if err != nil {
+		return err
+	}
+	w.mtx.Lock()
+	w.lockTime = time.Now().Add(dur)
+	w.mtx.Unlock()
+	return nil
+}
+
+// lock the wallet, setting the lockTime to the time when the wallet will be
+// locked.
+func (w *xcWallet) Lock() error {
```

Comment:
I think the only thing changed about `xcWallet` is that I've exported the `Lock` method. It's a method of the embedded type, and having a `lock` method and a `Lock` method makes it possible to bypass the `lockTime` handling by mistake. This change is meant to prevent that, and to bring consistency with the `Unlock` method, which already did this.